cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -107,3 +118,51 @@ def test_verify_revert_snapshot(test_vm, provider, soft_assert, register_event,
         logger.info('Revert to snapshot %s successful', snapshot1.name)
     except:
         logger.info('Revert to snapshot %s Failed', snapshot1.name)
+
+
+@pytest.mark.uncollectif(lambda provider: provider.type != 'virtualcenter')
+@pytest.mark.meta(blockers=[1247664], automates=[1247664])
+def test_create_snapshot_via_ae(request, domain, test_vm):
+    """This test checks whether the vm.create_snapshot works in AE.
+
+    Prerequisities:
+        * A VMware provider
+        * A VM that has been discovered by CFME
+
+    Steps:
+        * Clone the Request class inside the System namespace into a new domain
+        * Add a method named ``snapshot`` and insert the provided code there.
+        * Add an instance named ``snapshot`` and set the methd from previous step
+            as ``meth5``
+        * Run the simulation of the method against the VM, preferably setting
+            ``snap_name`` to something that can be checked
+        * Wait until snapshot with such name appears.
+    """
+    # PREPARE
+    file = data_path.join("ui").join("automate").join("test_create_snapshot_via_ae.rb")
+    with file.open("r") as f:
+        method_contents = f.read()
+    miq_domain = Domain("ManageIQ (Locked)")
```

Comment:
Though this is fine, we should probably build in something so that we don't have to say ```(Locked)``` each time. Locked or unlocked, the domain should still be found.